monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -94,14 +95,28 @@ def get_first_monkey_time(cls):
         return min(agents, key=lambda a: a.start_time).start_time
 
     @classmethod
-    def get_last_monkey_dead_time(cls):
-        agents = filter(lambda a: a.stop_time is not None, cls._agent_repository.get_agents())
+    def get_last_monkey_dead_time(cls) -> Optional[datetime]:
+        agents = cls._agent_repository.get_agents()
+        if not agents:
+            return None
+
+        # TODO: Make sure that the case where an agent doesn't have a stop time
+        #       because it doesn't send a shutdown event is solved after #2518.
+        #       Till then, if an agent doesn't have stop time, the total run duration
+        #       won't be shown in the report.
```

Comment:
So now we'll have to go back here and fix the code that doesn't belong here in the first place?

Reply:
We won't necessarily have to fix it, and this was an issue (it returned the monkey run duration even if agents were still running) that existed before this PR.